sdk/translation/Azure.AI.Translation.Document/tests/Infrastructure/DocumentTranslationLiveTestBase.cs

```diff
@@ -78,6 +90,22 @@ public async Task<Uri> CreateSourceContainerAsync(List<TestDocument> documents)
             return containerClient.GenerateSasUri(BlobContainerSasPermissions.List | BlobContainerSasPermissions.Read, expiresOn);
         }
 
+        public Uri CreateTargetContainer(List<TestDocument> documents = default)
+        {
+            Recording.DisableIdReuse();
```

Comment:
I wonder if a good way to avoid having the same logic in so many functions is to create 2 functions (one sync and one async) with logic:1. get a container name, create the container, upload the document, return the container`,
and then `source`, `target`, and `glossary` functions can use call those functions.
What do you think?

Reply:
Yes!! I was actually wondering why it wasn't set up this way originally (before I added glossaries), but I realized that the main difference was in the permissions provided when generating the Sas Uri. I can set those up as parameters as well such that a function would look like:

`public uri CreateContainerAsync(string containerName, List<TestDocument> , List<BlobContainerSasPermissions>)`

Would this be good?

Reply:
Yes! Sounds great. Will implement this first thing tomorrow.

Reply:
I somehow got a notification from you asking about the permissions. That is a good point.
We can make them a parameter too

Reply:
Oh yea, I initially had a comment about that but I replaced it thinking that I can just return the containerClient like you mentioned. Then the caller function can generate the Sas with whichever permissions it needs. Not sure which one is more elegant.

Reply:
I think both have pros and cons. whatever should be ok